sklearn/linear_model/_logistic.py

```diff
@@ -1762,6 +1764,8 @@ class LogisticRegressionCV(LogisticRegression, LinearClassifierMixin, BaseEstima
     0.98...
     """
 
+    # TODO(1.9): remove this when sample_weight is removed from the `score` signature
+    __metadata_request__score = {"sample_weight": metadata_routing.UNUSED}
```

Comment:
this is to avoid `set_score_request` be present on this class

Reply:
Ah okay! 

But currently in this PR, `set_score_request` does not raise if we pass `sample_weight`:
```python
model = LogisticRegressionCV(cv=5, max_iter=1000)
model.set_score_request(sample_weight=True)
get_routing_for_object(model)
{'$self_request': {'fit': {'sample_weight': None}, 'score': {'sample_weight': True}}, ...}
```
I would have assumed that the __metadata_request__score class default was to prevent `sample_weight` from showing up in the `LogisticRegressionCV.set_score_request` signature ?